src/Psalm/Type.php

```diff
@@ -110,9 +110,13 @@ public static function getStringFromFQCLN(
         ?string $namespace,
         array $aliased_classes,
         ?string $this_class,
-        bool $allow_self = false
+        bool $allow_self = false,
+        bool $was_static = false
     ) : string {
         if ($allow_self && $value === $this_class) {
+            if ($was_static) {
```

Comment:
The condition should be inverse?

Reply:
Why do you think so? This boolean is true when in static context so we add `$this`/`static` on the phpdoc when it's true